File: 0739-daily-temperatures/0739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    const res = Array(temperatures.length).fill(0);
    let stack = [temperatures.length - 1];
    for (let i = temperatures.length - 2; i >= 0; i--) {
        while (stack.length) {
            const last = stack.pop();
            if (temperatures[i] < temperatures[last]) {
                stack.push(last)
                res[i] = last - i;
                break;
            }
        }
        stack.push(i);
    }

    return res;
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0 for _ in range(len(temperatures))]
        i = len(temperatures) - 2
        stack = [len(temperatures) - 1]

        while i >= 0:
            while stack:
                last = stack.pop()
                if temperatures[i] < temperatures[last]:
                    stack.append(last)
                    res[i] = last - i
                    break
            stack.append(i)
            i -= 1
        
        return res
                    